### PR TITLE
[Perf/scale] Introduce web-burner step

### PR DESCRIPTION
**Pull request by josecastillolema**
Generic [web-burner](https://github.com/kube-burner/kube-burner-ocp/tree/main/cmd/config/web-burner-init) kube-burner-ocp step with cleanup. Covers both node-density and cluster-density.

With the current environment (6 workers) and object count some jobs execute too fast for the metric collection. Regardless, most of the relevant metrics are indeed collected and correctly presented on CPT dashboards & Grafana.
As we plan to gradually move to a bigger (virtualized) environment and the step is functionally correct I suggest we merge it while we work on other environment topologies in the deploy step.

### CODE REVIEW

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
cc @jtaleric

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-node-density-heavy-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
@jtaleric ptal

**Comment by krishvoor** [comment]
Seeing some issues seen during Indexing - etcd leader change observed: [URL](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/52150/rehearse-52150-pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes/1791473544228507648/artifacts/web-burner-node-density-6nodes/openshift-qe-web-burner-node-density/build-log.txt)
Will issue a re-run.

**Comment by krishvoor** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by vishnuchalla** [comment]
Looks like the rehearsal job is still failing. Might need a re-run for live debugging.

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-cluster-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-cluster-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-cluster-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-cluster-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

File: ci-operator/step-registry/openshift-qe/web-burner/openshift-qe-web-burner-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  - name: CHURN
+    default: "false"
+    documentation: |-
+      Default is false, which means the workload will not delete and recreate a set number of namespaces
+  - name: CRD
+    default: "false"
+    documentation: |-
+      Enable the new AdminPolicyBasedExternalRoute CR
+  - name: E2E_VERSION
+    default: "v2.1.7"
```
Comment (vishnuchalla) [inline ci-operator/step-registry/openshift-qe/web-burner/openshift-qe-web-burner-ref.yaml:25]: This needs an update to `v2.1.8` in order for the latest web-burner changes to get reflected.

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-node-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-cluster-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-cluster-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-cluster-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-cluster-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-cluster-density-6nodes

**Comment by josecastillolema** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-metal-4.15-nightly-x86-web-burner-cluster-density-6nodes

**Comment by josecastillolema** [comment]
@jtaleric @krishvoor ptal

**Review by krishvoor — APPROVED** [review]
Overall looks good - most recent job indicated issues with Indexing.

**Comment by jtaleric** [comment]
@josecastillolema looking at the latest run, it still appears to be executing too fast for the metric collection 

```
time="2024-06-19 15:39:58" level=info msg="🔍 Endpoint: https://prometheus-k8s-openshift-monitoring.apps.bm.example.com; profile: config/metrics-report.yml start: 2024-06-19T15:39:48Z end: 2024-06-19T15:39:48Z; job: create-networks-job" file="prometheus.go:70"
time="2024-06-19 15:39:58" level=warning msg="Error found with query avg_over_time(histogram_quantile(0.99, sum(irate(apiserver_request_duration_seconds_bucket{apiserver=\"kube-apiserver\", verb=~\"LIST|GET\", subresource!~\"log|exec|portforward|attach|proxy\"}[2m])) by (le, resource, verb, scope))[0s:]) > 0: bad_data: invalid parameter \"query\": 1:226: parse error: duration must be greater than 0" file="prometheus.go:201"
time="2024-06-19 15:39:58" level=warning msg="Error found with query avg_over_time(histogram_quantile(0.99, sum(irate(apiserver_request_duration_seconds_bucket{apiserver=\"kube-apiserver\", verb=~\"LIST|GET\", subresource!~\"log|exec|portforward|attach|proxy\"}[2m])) by (le, resource, verb, scope))[0s:]) > 0: bad_data: invalid parameter \"query\": 1:226: parse error: duration must be greater than 0" file="prometheus.go:201"
time="2024-06-19 15:39:58" level=warning msg="Error found with query max_over_time(histogram_quantile(0.99, sum(irate(apiserver_request_duration_seconds_bucket{apiserver=\"kube-apiserver\", verb=~\"LIST|GET\", subresource!~\"log|exec|portforward|attach|proxy\"}[2m])) by (le, resource, verb, scope))[0s:]) > 0: bad_data: invalid parameter \"query\": 1:226: parse error: duration must be greater than 0" file="prometheus.go:201"
time="2024-06-19 15:39:58" level=warning msg="Error found with query max_over_time(histogram_quantile(0.99, sum(irate(apiserver_request_duration_seconds_bucket{apiserver=\"kube-apiserver\", verb=~\"LIST|GET\", subresource!~\"log|exec|portforward|attach|proxy\"}[2m])) by (le, resource, verb, scope))[0s:]) > 0: bad_data: invalid parameter \"query\": 1:226: parse error: duration must be greater than 0" file="prometheus.go:201"
time="2024-06-19 15:39:58" level=warning msg="Error found with query avg_over_time(histogram_quantile(0.99, sum(irate(apiserver_request_duration_seconds_bucket{apiserver=\"kube-apiserver\", verb=~\"POST|PUT|DELETE|PATCH\", subresource!~\"log|exec|portforward|attach|proxy\"}[2m])) by (le, resource, verb, scope))[0s:]) > 0: bad_data: invalid parameter \"query\": 1:239: parse error: duration must be greater than 0" file="prometheus.go:201"
time="2024-06-19 15:39:58" level=warning msg="Error found with query avg_over_time(histogram_quantile(0.99, sum(irate(apiserver_request_duration_seconds_bucket{apiserver=\"kube-apiserver\", verb=~\"POST|PUT|DELETE|PATCH\", subresource!~\"log|exec|portforward|attach|proxy\"}[2m])) by (le, resource, verb, scope))[0s:]) > 0: bad_data: invalid parameter \"query\": 1:239: parse error: duration must be greater than 0" file="prometheus.go:201"
time="2024-06-19 15:39:58" level=warning msg="Error found with query max_over_time(histogram_quantile(0.99, sum(irate(apiserver_request_duration_seconds_bucket{apiserver=\"kube-apiserver\", verb=~\"POST|PUT|DELETE|PATCH\", subresource!~\"log|exec|portforward|attach|proxy\"}[2m])) by (le, resource, verb, scope))[0s:]) > 0: bad_data: invalid parameter \"query\": 1:239: parse error: duration must be greater than 0" file="prometheus.go:201"
time="2024-06-19 15:39:58" level=warning msg="Error found with query max_over_time(histogram_quantile(0.99, sum(irate(apiserver_request_duration_seconds_bucket{apiserver=\"kube-apiserver\", verb=~\"POST|PUT|DELETE|PATCH\", subresource!~\"log|exec|portforward|attach|proxy\"}[2m])) by (le, resource, verb, scope))[0s:]) > 0: bad_data: invalid parameter \"query\": 1:239: parse error: duration must be greater than 0" file="prometheus.go:201"
```

Should we : 
a) disable prom metric collection? 
b) extend the workload to run longer?

**Comment by jtaleric** [comment]
Chatted with @josecastillolema -- once he updates his PR description with our discussion, I will +1 this one!

**Comment by josecastillolema** [comment]
With the current environment (6 workers) and object count some jobs execute too fast for the metric collection. Regardless, most of the relevant metrics are indeed collected and correctly presented on CPT dashboards & Grafana.
As we plan to gradually move to a bigger (virtualized) environment and the step is functionally correct I suggest we merge it while we work on other environment topologies in the deploy step.

**Pull request merged**